### DISCUSSION TITLE
HOW TO READ & WRITE

### DISCUSSION THREAD

**Discussion by nageshdudam** [exceljs/exceljs#2510] [Q&A]
There is no single example of how to write a file and read it.

How to write to an existing xlsx file. Again no example.

It just contains explanation of doing beautification to xlsx.

No proper examples from start to end.

**Comment by Harshitg10** [comment]
this code worked for me
Reading XLSX

// read from a file give path to Excel/xlsx file
var workbook = new Excel.Workbook();
workbook.xlsx.readFile(filename)
    .then(function() {
        // use workbook
    });

Writing XLSX

// write to a file
var workbook = new Excel.Workbook();
workbook.xlsx.writeFile(filename)
    .then(function() {
        // done
    });

**Comment by nageshdudam** [comment]
Thank you Harshit.

**Comment by ChenWes** [comment]
very helpful , thanks!

**Comment by danielmlc** [comment]
the code dons't work for me
>`var workbook = new Excel.Workbook();
workbook.xlsx.readFile(filename)
.then(function() {
// use workbook
});`

**Comment by Harshitg10** [comment]
@danielmlc  could you please explain me, in what scenario you are using this code and is it giving any error???

**Comment by danielmlc** [comment]
this is my code and err. 
`
var workbook = new Excel.Workbook();
      workbook.xlsx.readFile(filename).then((work)=>{
        console.log('success!')
      }).catch(err=>{
        console.log(err);
      })
`

TypeError: fs.exists is not a function
    at eval (exceljs.js?74b2:4140)
    at lib$es6$promise$$internal$$initializePromise (exceljs.js?74b2:16692)
    at Promish.lib$es6$promise$promise$$Promise [as constructor] (exceljs.js?74b2:16983)
    at new Promish (exceljs.js?74b2:36322)
    at Object.exists (exceljs.js?74b2:4139)
    at module.exports.readFile (exceljs.js?74b2:10893)
    at VueComponent._change (HelloWorld.vue?25ba:30)
    at boundFn (vue.esm.js?5e8e:190)
    at invoker (vue.esm.js?5e8e:2004)
    at HTMLInputElement.fn._withTask.fn._withTask (vue.esm.js?5e8e:1802)

**Comment by Harshitg10** [comment]
@danielmlc  are you using it in frontend??

**Comment by danielmlc** [comment]
@Harshitg10 
 I using it in frontend
  How do I readFile  to the front end?

**Comment by Harshitg10** [comment]
@danielmlc Sorry Brother, but this is a NodeJS Library. You can't use it in frontend

**Comment by josocial** [comment]
@Harshitg10 Does this mean we can;t use the .readFile function on client side? I have a use case where is want to load an predefined Excel File and populate it with data. This is a purely client side operation and no service call involved. Is this possible?

**Comment by Harshitg10** [comment]
@josocial for viewing an excel in a read-only mode you can use Microsoft office 365 viewing api https://products.office.com/en-in/office-online/view-office-documents-online but your document should be available in public. My team is also looking for similar kind of solution where we can give user, an experience as if they are looking their document as they look in their native app. There are some library which can convert an excel file to html but look and feel won't be there.
